### PR TITLE
Sinon: Added callsFake method for SinonStub

### PR DESCRIPTION
**Pull request by vilmosioo**
Please fill in this template.

- [x] Make your PR against the `master` branch.
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped#common-mistakes).
- [x] Run `tsc` without errors.
- [ ] Run `npm run lint package-name` if a `tslint.json` is present.

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not provide its own types, and you can not add them.
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `npm run new-package package-name`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, and `strictNullChecks` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/sinonjs/sinon/blob/71214b497aaa1bd28e4d5262f1d2bca3c327b03e/docs/current/migrating-to-2.0.md
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "../tslint.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: sinon/index.d.ts [hunk 1]
```diff
@@ -109,6 +109,7 @@ declare namespace Sinon {
         callsArgOnAsync(index: number, context: any): SinonStub;
         callsArgWithAsync(index: number, ...args: any[]): SinonStub;
         callsArgOnWithAsync(index: number, context: any, ...args: any[]): SinonStub;
+        callsFake(func: any): SinonStub;
```
Comment (DanielRosenwasser) [inline sinon/index.d.ts]: Do you have a more accurate type for `func`? I'd pick `Function` or `(...args: any[] => void` at the very least.

Comment (vilmosioo) [inline sinon/index.d.ts]: No problem.

File: sinon/index.d.ts [hunk 2]
```diff
@@ -109,6 +109,7 @@ declare namespace Sinon {
         callsArgOnAsync(index: number, context: any): SinonStub;
         callsArgWithAsync(index: number, ...args: any[]): SinonStub;
         callsArgOnWithAsync(index: number, context: any, ...args: any[]): SinonStub;
+        callsFake(...args: any[]): SinonStub;
```
Comment (DanielRosenwasser) [inline sinon/index.d.ts]: I don't know if this is what I meant - does `callsFake` actually take multiple arguments? I figured that it was

```ts
callsFake(func: (...args: any[]) => void)
```

Comment (vilmosioo) [inline sinon/index.d.ts]: The argument is a callback function. The deprecated method had `func: any`. I'll change it for `func: (...args: any[]) => void`

**Comment by vilmosioo** [comment]
@DanielRosenwasser hey, is everything ok with this PR?

**Pull request merged**